cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_collaborators.spec.js

```diff
@@ -0,0 +1,211 @@
+      });
+    });
+
+    afterEach(() => {
+      cy.deleteWorkspaceById(workspaceId);
+    });
+
+    it('should add user and group collaborators successfully', () => {
+      //Add user
+      cy.getElementByTestId('add-collaborator-popover').click();
+      cy.get('button[id="user"]').click();
+      cy.contains('Add Users').should('be.visible');
+      cy.getElementByTestId('workspaceCollaboratorIdInput-0').type(
+        'new_read_user'
+      );
+      cy.get('button[type="submit"]')
+        .contains('span', 'Add collaborators')
+        .click();
+
+      //Add group
+      cy.getElementByTestId('add-collaborator-popover').click();
+      cy.get('button[id="group"]').click();
+      cy.contains('Add Groups').should('be.visible');
+      cy.getElementByTestId('workspaceCollaboratorIdInput-0').type(
+        'new_admin_group'
+      );
+      cy.get('span[title="Admin"]').click();
+      cy.get('button[type="submit"]')
+        .contains('span', 'Add collaborators')
+        .click();
+
+      cy.contains('new_read_user');
+
+      cy.get('table')
+        .contains('td', 'new_read_user')
+        .parent()
+        .within(() => {
+          cy.get('td').eq(3).contains('Read only');
+        });
+
+      cy.get('table')
+        .contains('td', 'new_admin_group')
+        .parent()
+        .within(() => {
+          cy.get('td').eq(3).contains('Admin');
+        });
+      const expectedWorkspace = {
+        name: workspaceName,
+        features: ['use-case-observability'],
+        description: 'test_description',
+        permissions: {
+          library_write: {
+            users: [`${Cypress.env('username')}`],
+            groups: ['admin_group', 'new_admin_group'],
+          },
+          write: {
+            users: [`${Cypress.env('username')}`],
+            groups: ['admin_group', 'new_admin_group'],
+          },
+          library_read: {
+            users: ['read_user', 'new_read_user'],
+          },
+          read: {
+            users: ['read_user', 'new_read_user'],
+          },
+        },
+      };
+      cy.checkWorkspace(workspaceId, expectedWorkspace);
+    });
+
+    it('should change access level successfully', () => {
+      cy.get('table')
+        .contains('td', 'read_user')
+        .parent('tr')
+        .within(() => {
+          cy.get('input[type="checkbox"]').check();
+        });
+      cy.get('table')
+        .contains('td', 'admin_group')
+        .parent('tr')
+        .within(() => {
+          cy.get('input[type="checkbox"]').check();
+        });
+
+      cy.getElementByTestId(
+        'workspace-detail-collaborator-table-actions'
+      ).click();
+      cy.get('div[role="dialog"]')
+        .find('span')
+        .contains('Change access level')
+        .click();
+      cy.get('div[role="dialog"]')
+        .find('span')
+        .contains('Read and write')
+        .click();
+      cy.getElementByTestId('confirmModalConfirmButton').click();
+
```

Comment:
nit: we can add a verification here to verify change access level toast exists

Reply:
I think we have verified the actual content in the table has been updated, this is more trustable than the toast.